custom/icds_reports/reports/governance_apis.py

```diff
@@ -38,3 +39,56 @@ def base_data(row_data):
     for row in paginated_data:
         data_rows.append(base_data(row))
     return data_rows, data.count()
+
+
+@icds_quickcache(['start', 'length', 'year', 'month', 'order', 'query_filters'], timeout=30 * 60)
+def get_beneficiary_data(start, length, year, month, order, query_filters):
+    data = AggAwcMonthly.objects.filter(
+        month=date(year, month, 1),
+        **query_filters
+    ).order_by(*order).values(
+        'month',
+        state=F('state_name'),
+        district=F('district_name'),
+        block=F('block_name'),
+        sector=F('supervisor_name'),
+        awc=F('awc_name'),
+        total_preg_benefit_till_date=F('cases_ccs_pregnant'),
+        total_lact_benefit_till_date=F('cases_ccs_lactating'),
+        total_preg_reg_till_date=F('cases_ccs_pregnant_all'),
+        total_lact_reg_till_date=F('cases_ccs_lactating_all'),
+        total_lact_benefit_in_month=F('cases_ccs_lactating_reg_in_month'),
+        total_preg_benefit_in_month=F('cases_ccs_pregnant_reg_in_month'),
+        total_lact_reg_in_month=F('cases_ccs_lactating_all_reg_in_month'),
+        total_preg_reg_in_month=F('cases_ccs_pregnant_all_reg_in_month'),
+        total_0_3_female_benefit_till_date=F('valid_all_0_3_female'),
+        total_0_3_male_benefit_till_date=F('valid_all_0_3_male'),
+        total_0_3_female_reg_till_date=F('open_all_0_3_female'),
+        total_0_3_male_reg_till_date=F('open_all_0_3_male'),
+        total_3_6_female_benefit_till_date=F('valid_all_3_6_female'),
+        total_3_6_male_benefit_till_date=F('valid_all_3_6_male'),
+        total_3_6_female_reg_till_date=F('open_all_3_6_female'),
+        total_3_6_male_reg_till_date=F('open_all_3_6_male'),
+        total_0_3_female_benefit_in_month=F('valid_reg_in_month_0_3_female'),
+        total_0_3_male_benefit_in_month=F('valid_reg_in_month_0_3_male'),
+        total_0_3_female_reg_in_month=F('open_reg_in_month_0_3_female'),
+        total_0_3_male_reg_in_month=F('open_reg_in_month_0_3_male'),
+        total_3_6_female_benefit_in_month=F('valid_reg_in_month_3_6_female'),
+        total_3_6_male_benfit_in_month=F('valid_reg_in_month_3_6_male'),
+        total_3_6_female_reg_in_month=F('open_reg_in_month_3_6_female'),
+        total_3_6_male_reg_in_month=F('open_reg_in_month_3_6_male'),
+
+    )
+
+    # To apply pagination on database query with offset and limit
+    paginated_data = data[int(start):(int(start) + length)]
```

Comment:
Not sure if I missed this in the last PR or didn't review the last PR, but this kind of pagination is very slow, especially as you get deep, and I imagine we expect all queries to try to eventually get the whole result set (longer description at https://use-the-index-luke.com/sql/partial-results/fetch-next-page). Ideally we would use the last record to find the next one, not an offset, but if this is only at the AWC level (and not beneficiary) it may be ok.